x-pack/plugins/security_solution/public/common/components/exceptions/viewer/exceptions_viewer_header.tsx

```diff
@@ -0,0 +1,181 @@
+        tags: foundTags.length ? foundTags.split(',') : [],
+      });
+    },
+    [onFiltersChange]
+  );
+
+  const onAddExceptionDropdownClick = useCallback(
+    (): void => setAddExceptionMenuOpen(!isAddExceptionMenuOpen),
+    [setAddExceptionMenuOpen]
+  );
+
+  const onAddException = useCallback(
+    (type: ToggleId): void => {
+      onAddExceptionClick(type);
+      setAddExceptionMenuOpen(false);
+    },
+    [onAddExceptionClick, setAddExceptionMenuOpen]
+  );
+
+  const getAddExceptionOptions = useMemo(
+    (): EuiContextMenuPanelDescriptor[] => [
+      {
+        id: 0,
+        items: [
+          {
+            name: i18n.ADD_TO_ENDPOINT_LIST,
+            onClick: () => {
+              onAddException(ToggleId.ENDPOINT);
+            },
+            'data-test-subj': 'addEndpointExceptionBtn',
+          },
+          {
+            name: i18n.ADD_TO_DETECTIONS_LIST,
+            onClick: () => {
+              onAddException(ToggleId.DETECTION_ENGINE);
+            },
+            'data-test-subj': 'addDetectionsExceptionBtn',
+          },
+        ],
+      },
+    ],
+    [onAddException]
+  );
+
+  const toggleOptions = useMemo((): EuiButtonGroupOption[] => {
+    return [
+      {
+        id: ToggleId.DETECTION_ENGINE,
+        label: 'Detection Engine',
+        isDisabled: !listTypes.includes(ToggleId.DETECTION_ENGINE),
+        'data-test-subj': 'detectionsToggle',
+      },
+      {
+        id: ToggleId.ENDPOINT,
+        label: 'Endpoint',
+        isDisabled: !listTypes.includes(ToggleId.ENDPOINT),
+        'data-test-subj': 'endpointToggle',
+      },
+    ];
+  }, [listTypes]);
+
+  return (
+    <EuiFlexGroup alignItems="center">
+      <EuiFlexItem grow={true}>
+        <EuiFieldSearch
+          aria-label={i18n.SEARCH_DEFAULT}
+          incremental={false}
+          placeholder={i18n.SEARCH_DEFAULT}
+          onChange={handleOnSearch}
+          disabled={isInitLoading || listTypes.length === 0}
+          fullWidth
+          data-test-subj="exceptionsHeaderSearch"
+        />
+      </EuiFlexItem>
+
+      <EuiFlexItem grow={false}>
+        <EuiButtonGroup
+          legend={'Exception list type toggle'}
+          options={toggleOptions}
+          idSelected={selectedListType}
+          onChange={onToggle}
+          isDisabled={isInitLoading}
+          data-test-subj="exceptionsHeaderListToggle"
+        />
+      </EuiFlexItem>
+
+      <EuiFlexItem grow={false}>
+        <EuiPopover
+          id="contextMenuNormal"
+          button={
+            <EuiButton
+              iconType="arrowDown"
+              iconSide="right"
+              onClick={onAddExceptionDropdownClick}
+              isDisabled={isInitLoading}
+              fill
+              data-test-subj="exceptionsHeaderAddExceptionBtn"
+            >
+              {i18n.ADD_EXCEPTION_LABEL}
+            </EuiButton>
+          }
+          isOpen={isAddExceptionMenuOpen}
+          closePopover={onAddExceptionDropdownClick}
```

Comment:
![image](https://user-images.githubusercontent.com/616158/83816840-55e8a480-a691-11ea-9de0-fc1aff8c1133.png)
The popover doesn't seem to be closing when clicking outside of it anywhere on the page.

Reply:
Thanks, I'll take a look!

Reply:
Probably want a `anchorPosition="downCenter"` here as well to match the other popovers. 

Related -- I think we'll want to have a conditional on whether or not a specific rule _supports_ lists of type `endpoint`, at least for this initial implementation. Then we can hide the filter buttons, and make this a single action button for just adding to `detections lists`. I say this as initially for 7.9 only the `Elastic Endpoint Rule` will have the ability to send exceptions to the endpoint, so we won't want to advertise this behavior on all rules just yet. I think it's fine to leave as is for now for testing purposes, but let's chat in the next sync about what we want to do here.

Reply:
Updated! @peluja1012 popover now closes clicking anywhere on screen. @spong I added a prop to get passed in that signals which list types are allowed.